src/ngircd/irc-info.c

```diff
@@ -817,8 +817,9 @@ IRC_NAMES( CLIENT *Client, REQUEST *Req )
 	}
 
 	/* Now print all clients which are not in any channel */
+	char chan_symbol = Channel_HasMode(chan, 's') ? '@' : '=';
```

Comment:
This will fail when Chan is NULL, and results in the test suite failing, because ngIRCd hits an assert():

```
[17006:5   37] User "test0!~user@127.0.0.1" registered (connection 7).
[17006:5   39] Got valid OPER for "TestOp" from "test0!~user@127.0.0.1", user is an IRC operator now.
Assertion failed: (Chan != NULL), function Channel_HasMode, file channel.c, line 531.
```

… and looking at the comment above this line and the code, the next `while()` loop iterates over all clients ans lists those, that are _not_ on any channel (`Channel_FirstChannelOf(c) == NULL`) – so I the "channel" character for "no channel" should always be "*", no? So basically only the changes in `IRC_Send_NAMES` are needed?

Reply:
how did you see this assertion failure?

Reply:
Basically with this sequence, ngIRCd must be build with debug code enabled:

```bash
./configure --enable-debug
make check
```
… seeing the test suite failing:
```
stressing server with 5 clients (be patient!):
checking stress script ... failure!
FAIL: stress-server.sh
stopping server 1 ... failure: no running server found!?
FAIL: stop-server1
```
… and the "no running server found" more or less translating to "must have crashed".

Then checking the logs:
```bash
tail src/testsuite/ngircd-test1.log
```